Scaled_joint_trajectory_controller does not respond to speed change on teach pendant

### Affected ROS2 Driver version(s)

Humble

### Used ROS distribution.

Humble

### Which combination of platform is the ROS driver running on.

Ubuntu Linux with standard kernel

### How is the UR ROS2 Driver installed.

Build both the ROS driver and UR Client Library from source

### Which robot platform is the driver connected to.

UR E-series robot

### Robot SW / URSim version(s)

5.13.0

### How is the ROS driver used.

Through the robot teach pendant using External Control URCap

### Issue details

# Summary
Controlling a UR10e robot with the URCap “External-Control” from an external PC. Waypoints are sent from the PC to the real robot, which it also follows, but the movement is aborted due to a time-out error as soon as the speed slider is not at 100%

## Issue details
We are trying to control a UR10e robot with Polyscope 5.13 via ROS2 Humble. The robot is connected to an external PC via Ethernet. The URCap “External-Control” is installed on the real robot. When I press the play button, I get the following output on the screen:
```
[ros2_control_node-1] [INFO] [1728380483.965475739] [UR_Client_Library:]: Robot requested program
[ros2_control_node-1] [INFO] [1728380483.965620055] [UR_Client_Library:]: Sent program to robot
[ros2_control_node-1] [INFO] [1728380484.260967978] [UR_Client_Library:]: Robot connected to reverse interface. Ready to receive control commands.
```
The Scaled_joint_trajectory_controller is active and receives waypoints via a C++ script. Everything works well and the robot follows the path. However, this is only possible if the speed controller is set to 100%. If I slow it down, I get the following error:
```
MoveGroupInterface::execute() failed or timeout reached
```
The robot moves to the first points of the path and then aborts the movement with the MoveGroup Error.

## Steps to Reproduce
Start the Launch-Files``` ros2 launch ur_robot_driver ur_control.launch.py ur_type:=ur10e robot_ip:=192.168.1.102 launch_rviz:=false ``` and ```ros2 launch ur_moveit_config ur_moveit.launch.py ur_type:=ur10e```

## Expected Behavior
The robot will slow down his movements, when the speed slider on the teach pendant is scaled down but don't aborted the movement.

## Actual Behavior
The robot moves to the first few points and then aborted the movement.

# Workaround Suggestion
Maybe I misunderstand the scaled_joint_trajectory_controller, but I thought this was a special controller for the UR series that takes into account the current value of the speed controller. I can also read the corresponding topic (```ros2 topic echo /speed_scaling_state_broadcaster/speed_scaling```) and get the same values as on the teach pendant. What other configurations need to be made on the controller? I have checked the ```ur_controllers.yaml``` and ```moveit_controllers.yaml```, but I have no idea where the problem is coming from.

Thanks in advance for any helpful tips!


### Relevant log output

```shell
cobot@MBS-Cobot:~$ ros2 launch ur_robot_driver ur10e_control_moveit.launch.py
[INFO] [launch]: All log files can be found below /home/cobot/.ros/log/2024-10-08-14-38-46-430353-MBS-Cobot-94879
[INFO] [launch]: Default logging verbosity is set to INFO
WARNING:root:Cannot infer URDF from `/home/cobot/lerosh_ws/install/ur_moveit_config/share/ur_moveit_config`. -- using config/ur.urdf
WARNING:root:Cannot infer SRDF from `/home/cobot/lerosh_ws/install/ur_moveit_config/share/ur_moveit_config`. -- using config/ur.srdf
WARNING:root:"File /home/cobot/lerosh_ws/install/ur_moveit_config/share/ur_moveit_config/config/ur.urdf doesn't exist"
WARNING:root:The robot description will be loaded from /robot_description topic 
[INFO] [dashboard_client-2]: process started with pid [94884]
[INFO] [controller_stopper_node-4]: process started with pid [94886]
[INFO] [ros2_control_node-1]: process started with pid [94883]
[INFO] [tool_communication.py-3]: process started with pid [94885]
[INFO] [urscript_interface-5]: process started with pid [94887]
[INFO] [robot_state_publisher-6]: process started with pid [94888]
[INFO] [spawner-7]: process started with pid [94889]
[INFO] [spawner-8]: process started with pid [94890]
[INFO] [spawner-9]: process started with pid [94891]
[INFO] [wait_for_robot_description-10]: process started with pid [94892]
[dashboard_client-2] [INFO] [1728391127.108349645] [UR_Client_Library:]: Connected: Universal Robots Dashboard Server
[dashboard_client-2] 
[controller_stopper_node-4] [INFO] [1728391127.154770768] [Controller stopper]: Waiting for switch controller service to come up on controller_manager/switch_controller
[robot_state_publisher-6] [INFO] [1728391127.172917546] [robot_state_publisher]: Robot initialized
[ros2_control_node-1] [INFO] [1728391127.203478408] [controller_manager]: Subscribing to '/robot_description' topic for robot description.
[ros2_control_node-1] [INFO] [1728391127.203838481] [controller_manager]: update rate is 500 Hz
[ros2_control_node-1] [WARN] [1728391127.204612930] [controller_manager]: Could not enable FIFO RT scheduling policy. Consider setting up your user to do FIFO RT scheduling. See [https://control.ros.org/master/doc/ros2_control/controller_manager/doc/userdoc.html] for details.
[ros2_control_node-1] [INFO] [1728391127.218509716] [controller_manager]: Received robot description from topic.
[ros2_control_node-1] text not specified in the tf_prefix tag
[ros2_control_node-1] [INFO] [1728391127.224958797] [resource_manager]: Loading hardware 'ur10e' 
[ros2_control_node-1] [INFO] [1728391127.230828177] [resource_manager]: Loaded hardware 'ur10e' from plugin 'ur_robot_driver/URPositionHardwareInterface'
[ros2_control_node-1] [INFO] [1728391127.231143896] [resource_manager]: Initialize hardware 'ur10e' 
[ros2_control_node-1] [INFO] [1728391127.231263148] [resource_manager]: Successful initialization of hardware 'ur10e'
[ros2_control_node-1] [INFO] [1728391127.232749616] [resource_manager]: 'configure' hardware 'ur10e' 
[ros2_control_node-1] [INFO] [1728391127.232797648] [URPositionHardwareInterface]: Starting ...please wait...
[ros2_control_node-1] [INFO] [1728391127.232851339] [URPositionHardwareInterface]: Initializing driver...
[ros2_control_node-1] [WARN] [1728391127.236470961] [UR_Client_Library:]: Your system/user seems not to be setup for FIFO scheduling. We recommend using a lowlatency kernel with FIFO scheduling. See https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/blob/master/ur_robot_driver/doc/real_time.md for details.
[tool_communication.py-3] [INFO] [1728391127.643704672] [ur_tool_comm]: 192.168.1.102
[tool_communication.py-3] [INFO] [1728391127.645903097] [ur_tool_comm]: Remote device is available at '/tmp/ttyTool2'
[tool_communication.py-3] [INFO] [1728391127.647115303] [ur_tool_comm]: Starting socat with following command:
[tool_communication.py-3] socat pty,link=/tmp/ttyTool2,raw,ignoreeof,waitslave tcp:192.168.1.102:54321
[ros2_control_node-1] [INFO] [1728391127.807475560] [UR_Client_Library:]: Negotiated RTDE protocol version to 2.
[ros2_control_node-1] [INFO] [1728391127.808167700] [UR_Client_Library:]: Setting up RTDE communication with frequency 500.000000
[wait_for_robot_description-10] [INFO] [1728391127.839388897] [wait_for_robot_description]: Waiting for message on /robot_description.
[wait_for_robot_description-10] [INFO] [1728391127.840446556] [wait_for_robot_description]: Received message on /robot_description. Shutting down.
[INFO] [wait_for_robot_description-10]: process has finished cleanly [pid 94892]
[INFO] [move_group-11]: process started with pid [95045]
[move_group-11] [INFO] [1728391128.551690396] [move_group.moveit.moveit.ros.rdf_loader]: Loaded robot model in 0.0342481 seconds
[move_group-11] [INFO] [1728391128.551852625] [move_group.moveit.moveit.core.robot_model]: Loading robot model 'ur10e'...
[move_group-11] [INFO] [1728391128.551888238] [move_group.moveit.moveit.core.robot_model]: No root/virtual joint specified in SRDF. Assuming fixed joint
[move_group-11] [INFO] [1728391128.572640940] [move_group.moveit.moveit.kinematics.kdl_kinematics_plugin]: Joint weights for group 'ur_manipulator': 1 1 1 1 1 1
[move_group-11] [INFO] [1728391128.645362003] [move_group.moveit.moveit.ros.planning_scene_monitor]: Publishing maintained planning scene on 'monitored_planning_scene'
[move_group-11] [INFO] [1728391128.645522853] [move_group.moveit.moveit.ros.moveit_cpp]: Listening to 'joint_states' for joint states
[move_group-11] [INFO] [1728391128.646679656] [move_group.moveit.moveit.ros.current_state_monitor]: Listening to joint states on topic 'joint_states'
[move_group-11] [INFO] [1728391128.647384193] [move_group.moveit.moveit.ros.planning_scene_monitor]: Listening to '/attached_collision_object' for attached collision objects
[move_group-11] [INFO] [1728391128.647410957] [move_group.moveit.moveit.ros.planning_scene_monitor]: Stopping existing planning scene publisher.
[move_group-11] [INFO] [1728391128.647541340] [move_group.moveit.moveit.ros.planning_scene_monitor]: Stopped publishing maintained planning scene.
[move_group-11] [INFO] [1728391128.648288897] [move_group.moveit.moveit.ros.planning_scene_monitor]: Publishing maintained planning scene on 'monitored_planning_scene'
[move_group-11] [INFO] [1728391128.648437020] [move_group.moveit.moveit.ros.planning_scene_monitor]: Starting planning scene monitor
[move_group-11] [INFO] [1728391128.649682948] [move_group.moveit.moveit.ros.planning_scene_monitor]: Listening to '/planning_scene'
[move_group-11] [INFO] [1728391128.649726017] [move_group.moveit.moveit.ros.planning_scene_monitor]: Starting world geometry update monitor for collision objects, attached objects, octomap updates.
[move_group-11] [INFO] [1728391128.650509238] [move_group.moveit.moveit.ros.planning_scene_monitor]: Listening to 'collision_object'
[move_group-11] [INFO] [1728391128.651177119] [move_group.moveit.moveit.ros.planning_scene_monitor]: Listening to 'planning_scene_world' for planning scene world geometry
[move_group-11] [WARN] [1728391128.652042255] [move_group.moveit.moveit.ros.occupancy_map_monitor]: Resolution not specified for Octomap. Assuming resolution = 0.1 instead
[move_group-11] [ERROR] [1728391128.652076477] [move_group.moveit.moveit.ros.occupancy_map_monitor]: No 3D sensor plugin(s) defined for octomap updates
[move_group-11] [INFO] [1728391128.668928029] [move_group.moveit.moveit.planners.pilz.joint_limits_aggregator]: Reading limits from namespace robot_description_planning
[move_group-11] [INFO] [1728391128.677797600] [move_group.moveit.moveit.planners.pilz.command_planner]: Available plugins: pilz_industrial_motion_planner/PlanningContextLoaderCIRC pilz_industrial_motion_planner/PlanningContextLoaderLIN pilz_industrial_motion_planner/PlanningContextLoaderPTP 
[move_group-11] [INFO] [1728391128.677842944] [move_group.moveit.moveit.planners.pilz.command_planner]: About to load: pilz_industrial_motion_planner/PlanningContextLoaderCIRC
[move_group-11] [INFO] [1728391128.680068002] [move_group.moveit.moveit.planners.pilz.command_planner]: Registered Algorithm [CIRC]
[move_group-11] [INFO] [1728391128.680096595] [move_group.moveit.moveit.planners.pilz.command_planner]: About to load: pilz_industrial_motion_planner/PlanningContextLoaderLIN
[move_group-11] [INFO] [1728391128.681890500] [move_group.moveit.moveit.planners.pilz.command_planner]: Registered Algorithm [LIN]
[move_group-11] [INFO] [1728391128.681925318] [move_group.moveit.moveit.planners.pilz.command_planner]: About to load: pilz_industrial_motion_planner/PlanningContextLoaderPTP
[move_group-11] [INFO] [1728391128.683286776] [move_group.moveit.moveit.planners.pilz.command_planner]: Registered Algorithm [PTP]
[move_group-11] [INFO] [1728391128.683316449] [move_group.moveit.moveit.ros.planning_pipeline]: Successfully loaded planner 'Pilz Industrial Motion Planner'
[move_group-11] [INFO] [1728391128.688096960] [move_group]: Try loading adapter 'default_planning_request_adapters/ResolveConstraintFrames'
[move_group-11] [INFO] [1728391128.689758753] [move_group]: Loaded adapter 'default_planning_request_adapters/ResolveConstraintFrames'
[move_group-11] [INFO] [1728391128.689780771] [move_group]: Try loading adapter 'default_planning_request_adapters/ValidateWorkspaceBounds'
[move_group-11] [INFO] [1728391128.690098293] [move_group]: Loaded adapter 'default_planning_request_adapters/ValidateWorkspaceBounds'
[move_group-11] [INFO] [1728391128.690117955] [move_group]: Try loading adapter 'default_planning_request_adapters/CheckStartStateBounds'
[move_group-11] [INFO] [1728391128.690157176] [move_group]: Loaded adapter 'default_planning_request_adapters/CheckStartStateBounds'
[move_group-11] [INFO] [1728391128.690173698] [move_group]: Try loading adapter 'default_planning_request_adapters/CheckStartStateCollision'
[move_group-11] [INFO] [1728391128.690207356] [move_group]: Loaded adapter 'default_planning_request_adapters/CheckStartStateCollision'
[move_group-11] [INFO] [1728391128.694305241] [move_group]: Try loading adapter 'default_planning_response_adapters/ValidateSolution'
[move_group-11] [INFO] [1728391128.702176169] [move_group]: Loaded adapter 'default_planning_response_adapters/ValidateSolution'
[move_group-11] [INFO] [1728391128.702218233] [move_group]: Try loading adapter 'default_planning_response_adapters/DisplayMotionPath'
[move_group-11] [INFO] [1728391128.703153725] [move_group]: Loaded adapter 'default_planning_response_adapters/DisplayMotionPath'
[move_group-11] [INFO] [1728391128.729455992] [move_group.moveit.moveit.ros.planning_pipeline]: Successfully loaded planner 'OMPL'
[move_group-11] [INFO] [1728391128.734626804] [move_group]: Try loading adapter 'default_planning_request_adapters/ResolveConstraintFrames'
[move_group-11] [INFO] [1728391128.735038982] [move_group]: Loaded adapter 'default_planning_request_adapters/ResolveConstraintFrames'
[move_group-11] [INFO] [1728391128.735057764] [move_group]: Try loading adapter 'default_planning_request_adapters/ValidateWorkspaceBounds'
[move_group-11] [INFO] [1728391128.735384128] [move_group]: Loaded adapter 'default_planning_request_adapters/ValidateWorkspaceBounds'
[move_group-11] [INFO] [1728391128.735404214] [move_group]: Try loading adapter 'default_planning_request_adapters/CheckStartStateBounds'
[move_group-11] [INFO] [1728391128.735449913] [move_group]: Loaded adapter 'default_planning_request_adapters/CheckStartStateBounds'
[move_group-11] [INFO] [1728391128.735467847] [move_group]: Try loading adapter 'default_planning_request_adapters/CheckStartStateCollision'
[move_group-11] [INFO] [1728391128.735495978] [move_group]: Loaded adapter 'default_planning_request_adapters/CheckStartStateCollision'
[move_group-11] [INFO] [1728391128.740242497] [move_group]: Try loading adapter 'default_planning_response_adapters/AddTimeOptimalParameterization'
[move_group-11] [INFO] [1728391128.741544208] [move_group]: Loaded adapter 'default_planning_response_adapters/AddTimeOptimalParameterization'
[move_group-11] [INFO] [1728391128.741631380] [move_group]: Try loading adapter 'default_planning_response_adapters/ValidateSolution'
[move_group-11] [INFO] [1728391128.742437678] [move_group]: Loaded adapter 'default_planning_response_adapters/ValidateSolution'
[move_group-11] [INFO] [1728391128.742465743] [move_group]: Try loading adapter 'default_planning_response_adapters/DisplayMotionPath'
[move_group-11] [INFO] [1728391128.743087942] [move_group]: Loaded adapter 'default_planning_response_adapters/DisplayMotionPath'
[move_group-11] [INFO] [1728391128.754218291] [move_group.moveit.moveit.ros.planning_pipeline]: Successfully loaded planner 'CHOMP'
[move_group-11] [INFO] [1728391128.758335178] [move_group]: Try loading adapter 'default_planning_request_adapters/ResolveConstraintFrames'
[move_group-11] [INFO] [1728391128.759078381] [move_group]: Loaded adapter 'default_planning_request_adapters/ResolveConstraintFrames'
[move_group-11] [INFO] [1728391128.759126613] [move_group]: Try loading adapter 'default_planning_request_adapters/ValidateWorkspaceBounds'
[move_group-11] [INFO] [1728391128.759751835] [move_group]: Loaded adapter 'default_planning_request_adapters/ValidateWorkspaceBounds'
[move_group-11] [INFO] [1728391128.759791041] [move_group]: Try loading adapter 'default_planning_request_adapters/CheckStartStateBounds'
[move_group-11] [INFO] [1728391128.759861181] [move_group]: Loaded adapter 'default_planning_request_adapters/CheckStartStateBounds'
[move_group-11] [INFO] [1728391128.759888123] [move_group]: Try loading adapter 'default_planning_request_adapters/CheckStartStateCollision'
[move_group-11] [INFO] [1728391128.759931713] [move_group]: Loaded adapter 'default_planning_request_adapters/CheckStartStateCollision'
[move_group-11] [INFO] [1728391128.764479924] [move_group]: Try loading adapter 'default_planning_response_adapters/AddTimeOptimalParameterization'
[move_group-11] [INFO] [1728391128.765582024] [move_group]: Loaded adapter 'default_planning_response_adapters/AddTimeOptimalParameterization'
[move_group-11] [INFO] [1728391128.765623137] [move_group]: Try loading adapter 'default_planning_response_adapters/ValidateSolution'
[move_group-11] [INFO] [1728391128.766377462] [move_group]: Loaded adapter 'default_planning_response_adapters/ValidateSolution'
[move_group-11] [INFO] [1728391128.766410942] [move_group]: Try loading adapter 'default_planning_response_adapters/DisplayMotionPath'
[move_group-11] [INFO] [1728391128.767016442] [move_group]: Loaded adapter 'default_planning_response_adapters/DisplayMotionPath'
[move_group-11] [INFO] [1728391128.776140314] [move_group.moveit.moveit.ros.planning_pipeline]: Successfully loaded planner 'STOMP'
[move_group-11] [INFO] [1728391128.780999040] [move_group]: Try loading adapter 'default_planning_request_adapters/ResolveConstraintFrames'
[move_group-11] [INFO] [1728391128.781471590] [move_group]: Loaded adapter 'default_planning_request_adapters/ResolveConstraintFrames'
[move_group-11] [INFO] [1728391128.781489050] [move_group]: Try loading adapter 'default_planning_request_adapters/ValidateWorkspaceBounds'
[move_group-11] [INFO] [1728391128.781918878] [move_group]: Loaded adapter 'default_planning_request_adapters/ValidateWorkspaceBounds'
[move_group-11] [INFO] [1728391128.781942790] [move_group]: Try loading adapter 'default_planning_request_adapters/CheckStartStateBounds'
[move_group-11] [INFO] [1728391128.781990352] [move_group]: Loaded adapter 'default_planning_request_adapters/CheckStartStateBounds'
[move_group-11] [INFO] [1728391128.782007146] [move_group]: Try loading adapter 'default_planning_request_adapters/CheckStartStateCollision'
[move_group-11] [INFO] [1728391128.782039129] [move_group]: Loaded adapter 'default_planning_request_adapters/CheckStartStateCollision'
[move_group-11] [INFO] [1728391128.786876315] [move_group]: Try loading adapter 'default_planning_response_adapters/AddTimeOptimalParameterization'
[move_group-11] [INFO] [1728391128.788217545] [move_group]: Loaded adapter 'default_planning_response_adapters/AddTimeOptimalParameterization'
[move_group-11] [INFO] [1728391128.788265435] [move_group]: Try loading adapter 'default_planning_response_adapters/ValidateSolution'
[move_group-11] [INFO] [1728391128.789055028] [move_group]: Loaded adapter 'default_planning_response_adapters/ValidateSolution'
[move_group-11] [INFO] [1728391128.789085716] [move_group]: Try loading adapter 'default_planning_response_adapters/DisplayMotionPath'
[move_group-11] [INFO] [1728391128.789710200] [move_group]: Loaded adapter 'default_planning_response_adapters/DisplayMotionPath'
[ros2_control_node-1] [INFO] [1728391128.850400117] [URPositionHardwareInterface]: Calibration checksum: 'calib_5119701370761913513'.
[move_group-11] [INFO] [1728391128.863610021] [move_group.moveit.moveit.plugins.simple_controller_manager]: Added FollowJointTrajectory controller for scaled_joint_trajectory_controller
[move_group-11] [INFO] [1728391128.867283599] [move_group.moveit.moveit.plugins.simple_controller_manager]: Added FollowJointTrajectory controller for joint_trajectory_controller
[move_group-11] [INFO] [1728391128.867495332] [move_group.moveit.moveit.plugins.simple_controller_manager]: Returned 2 controllers in list
[move_group-11] [INFO] [1728391128.867547925] [move_group.moveit.moveit.plugins.simple_controller_manager]: Returned 2 controllers in list
[move_group-11] [INFO] [1728391128.868156074] [move_group.moveit.moveit.ros.trajectory_execution_manager]: Trajectory execution is managing controllers
[move_group-11] [INFO] [1728391128.868183659] [move_group]: MoveGroup debug mode is ON
[move_group-11] [INFO] [1728391128.901499641] [move_group.moveit.moveit.planners.pilz.move_group_sequence_action]: initialize move group sequence action
[move_group-11] [INFO] [1728391128.905717310] [move_group.moveit.moveit.planners.pilz.joint_limits_aggregator]: Reading limits from namespace robot_description_planning
[move_group-11] [INFO] [1728391128.905947033] [move_group.moveit.moveit.planners.pilz.joint_limits_aggregator]: Reading limits from namespace robot_description_planning
[move_group-11] [INFO] [1728391128.907072523] [move_group.moveit.moveit.ros.move_group.executable]: 
[move_group-11] 
[move_group-11] ********************************************************
[move_group-11] * MoveGroup using: 
[move_group-11] *     - apply_planning_scene_service
[move_group-11] *     - clear_octomap_service
[move_group-11] *     - get_group_urdf
[move_group-11] *     - CartesianPathService
[move_group-11] *     - execute_trajectory_action
[move_group-11] *     - get_planning_scene_service
[move_group-11] *     - kinematics_service
[move_group-11] *     - move_action
[move_group-11] *     - motion_plan_service
[move_group-11] *     - query_planners_service
[move_group-11] *     - state_validation_service
[move_group-11] *     - SequenceAction
[move_group-11] *     - SequenceService
[move_group-11] ********************************************************
[move_group-11] 
[move_group-11] [INFO] [1728391128.907132256] [move_group.moveit.moveit.ros.move_group.context]: MoveGroup context using pipeline ompl
[move_group-11] [INFO] [1728391128.907164357] [move_group.moveit.moveit.ros.move_group.context]: MoveGroup context initialization complete
[move_group-11] Loading 'move_group/ApplyPlanningSceneService'...
[move_group-11] Loading 'move_group/ClearOctomapService'...
[move_group-11] Loading 'move_group/GetUrdfService'...
[move_group-11] Loading 'move_group/MoveGroupCartesianPathService'...
[move_group-11] Loading 'move_group/MoveGroupExecuteTrajectoryAction'...
[move_group-11] Loading 'move_group/MoveGroupGetPlanningSceneService'...
[move_group-11] Loading 'move_group/MoveGroupKinematicsService'...
[move_group-11] Loading 'move_group/MoveGroupMoveAction'...
[move_group-11] Loading 'move_group/MoveGroupPlanService'...
[move_group-11] Loading 'move_group/MoveGroupQueryPlannersService'...
[move_group-11] Loading 'move_group/MoveGroupStateValidationService'...
[move_group-11] Loading 'pilz_industrial_motion_planner/MoveGroupSequenceAction'...
[move_group-11] Loading 'pilz_industrial_motion_planner/MoveGroupSequenceService'...
[move_group-11] 
[move_group-11] You can start planning now!
[move_group-11] 
[spawner-8] [INFO] [1728391129.777277521] [spawner_joint_state_broadcaster]: Waiting for '/controller_manager' services to be available
[spawner-7] [INFO] [1728391129.841637447] [spawner_scaled_joint_trajectory_controller]: Waiting for '/controller_manager' services to be available
[ros2_control_node-1] [ERROR] [1728391129.900536124] [URPositionHardwareInterface]: The calibration parameters of the connected robot don't match the ones from the given kinematics config file. Please be aware that this can lead to critical inaccuracies of tcp positions. Use the ur_calibration tool to extract the correct calibration from the robot and pass that into the description. See [https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/blob/main/ur_calibration/README.md] for details.
[ros2_control_node-1] [INFO] [1728391129.900815833] [URPositionHardwareInterface]: System successfully started!
[ros2_control_node-1] [INFO] [1728391129.900879839] [resource_manager]: Successful 'configure' of hardware 'ur10e'
[ros2_control_node-1] [INFO] [1728391129.901383433] [resource_manager]: 'activate' hardware 'ur10e' 
[ros2_control_node-1] [INFO] [1728391129.901432618] [URPositionHardwareInterface]: Activating HW interface
[ros2_control_node-1] [INFO] [1728391129.901444066] [resource_manager]: Successful 'activate' of hardware 'ur10e'
[ros2_control_node-1] [WARN] [1728391129.904161024] [UR_Client_Library:]: Your system/user seems not to be setup for FIFO scheduling. We recommend using a lowlatency kernel with FIFO scheduling. See https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/blob/master/ur_robot_driver/doc/real_time.md for details.
[controller_stopper_node-4] [INFO] [1728391129.919669683] [Controller stopper]: Service available
[controller_stopper_node-4] [INFO] [1728391129.919729313] [Controller stopper]: Waiting for list controllers service to come up on controller_manager/list_controllers
[controller_stopper_node-4] [INFO] [1728391129.919769903] [Controller stopper]: Service available
[ros2_control_node-1] [INFO] [1728391129.942362874] [controller_manager]: Loading controller 'forward_position_controller'
[spawner-9] [INFO] [1728391129.964601485] [spawner_forward_position_controller]: Loaded forward_position_controller
[ros2_control_node-1] [INFO] [1728391129.966257929] [controller_manager]: Configuring controller 'forward_position_controller'
[ros2_control_node-1] [INFO] [1728391129.968194337] [forward_position_controller]: configure successful
[ros2_control_node-1] [INFO] [1728391129.989418217] [controller_manager]: Loading controller 'joint_state_broadcaster'
[spawner-8] [INFO] [1728391130.012558077] [spawner_joint_state_broadcaster]: Loaded joint_state_broadcaster
[ros2_control_node-1] [INFO] [1728391130.014077403] [controller_manager]: Configuring controller 'joint_state_broadcaster'
[ros2_control_node-1] [INFO] [1728391130.014256237] [joint_state_broadcaster]: 'joints' or 'interfaces' parameter is empty. All available state interfaces will be published
[spawner-8] [INFO] [1728391130.031030111] [spawner_joint_state_broadcaster]: Configured and activated joint_state_broadcaster
[ros2_control_node-1] [INFO] [1728391130.035200559] [controller_manager]: Loading controller 'io_and_status_controller'
[ros2_control_node-1] [INFO] [1728391130.063446754] [controller_manager]: Loading controller 'scaled_joint_trajectory_controller'
[spawner-8] [INFO] [1728391130.063646844] [spawner_joint_state_broadcaster]: Loaded io_and_status_controller
[ros2_control_node-1] [INFO] [1728391130.089213059] [controller_manager]: Configuring controller 'io_and_status_controller'
[spawner-7] [INFO] [1728391130.090895923] [spawner_scaled_joint_trajectory_controller]: Loaded scaled_joint_trajectory_controller
[ros2_control_node-1] [INFO] [1728391130.092713232] [controller_manager]: Configuring controller 'scaled_joint_trajectory_controller'
[ros2_control_node-1] [INFO] [1728391130.093373120] [scaled_joint_trajectory_controller]: No specific joint names are used for command interfaces. Using 'joints' parameter.
[ros2_control_node-1] [INFO] [1728391130.093605955] [scaled_joint_trajectory_controller]: Command interfaces are [position] and state interfaces are [position velocity].
[ros2_control_node-1] [INFO] [1728391130.093682455] [scaled_joint_trajectory_controller]: Using 'splines' interpolation method.
[ros2_control_node-1] [INFO] [1728391130.098170640] [scaled_joint_trajectory_controller]: Action status changes will be monitored at 20.00 Hz.
[spawner-8] [INFO] [1728391130.129911743] [spawner_joint_state_broadcaster]: Configured and activated io_and_status_controller
[spawner-7] [INFO] [1728391130.133290569] [spawner_scaled_joint_trajectory_controller]: Configured and activated scaled_joint_trajectory_controller
[ros2_control_node-1] [INFO] [1728391130.134756124] [controller_manager]: Loading controller 'speed_scaling_state_broadcaster'
[INFO] [spawner-9]: process has finished cleanly [pid 94891]
[ros2_control_node-1] [INFO] [1728391130.151169462] [speed_scaling_state_broadcaster]: Loading UR SpeedScalingStateBroadcaster with tf_prefix: 
[spawner-8] [INFO] [1728391130.156187964] [spawner_joint_state_broadcaster]: Loaded speed_scaling_state_broadcaster
[ros2_control_node-1] [INFO] [1728391130.158451223] [controller_manager]: Configuring controller 'speed_scaling_state_broadcaster'
[ros2_control_node-1] [INFO] [1728391130.158620357] [speed_scaling_state_broadcaster]: Publisher rate set to : 100.0 Hz
[spawner-8] [INFO] [1728391130.169530409] [spawner_joint_state_broadcaster]: Configured and activated speed_scaling_state_broadcaster
[ros2_control_node-1] [INFO] [1728391130.175772237] [controller_manager]: Loading controller 'force_torque_sensor_broadcaster'
[spawner-8] [INFO] [1728391130.199331945] [spawner_joint_state_broadcaster]: Loaded force_torque_sensor_broadcaster
[ros2_control_node-1] [INFO] [1728391130.201427300] [controller_manager]: Configuring controller 'force_torque_sensor_broadcaster'
[spawner-8] [INFO] [1728391130.213443778] [spawner_joint_state_broadcaster]: Configured and activated force_torque_sensor_broadcaster
[INFO] [spawner-7]: process has finished cleanly [pid 94889]
[INFO] [spawner-8]: process has finished cleanly [pid 94890]
[ros2_control_node-1] [INFO] [1728391161.605165820] [UR_Client_Library:]: Robot requested program
[ros2_control_node-1] [INFO] [1728391161.605328280] [UR_Client_Library:]: Sent program to robot
[ros2_control_node-1] [INFO] [1728391161.884950396] [UR_Client_Library:]: Robot connected to reverse interface. Ready to receive control commands.
```


### Accept Public visibility

- [X] I agree to make this context public

--- a/ur_moveit_config/launch/ur_moveit.launch.py
+++ b/ur_moveit_config/launch/ur_moveit.launch.py
@@ -188,7 +188,7 @@ def launch_setup(context, *args, **kwargs):
         "trajectory_execution.allowed_execution_duration_scaling": 1.2,
         "trajectory_execution.allowed_goal_duration_margin": 0.5,
         "trajectory_execution.allowed_start_tolerance": 0.01,
-        # Execution time monitoring doesn't make sense with the scaled JTC
+        # Execution time monitoring can be incompatible with the scaled JTC
         "trajectory_execution.execution_duration_monitoring": False,
     }
 

--- a/ur_moveit_config/launch/ur_moveit.launch.py
+++ b/ur_moveit_config/launch/ur_moveit.launch.py
@@ -188,6 +188,8 @@ def launch_setup(context, *args, **kwargs):
         "trajectory_execution.allowed_execution_duration_scaling": 1.2,
         "trajectory_execution.allowed_goal_duration_margin": 0.5,
         "trajectory_execution.allowed_start_tolerance": 0.01,
+        # Execution time monitoring doesn't make sense with the scaled JTC
+        "trajectory_execution.execution_duration_monitoring": False,
     }
 
     planning_scene_monitor_parameters = {

--- a/ur_robot_driver/doc/usage.rst
+++ b/ur_robot_driver/doc/usage.rst
@@ -192,11 +192,16 @@ Now you should be able to use the MoveIt Plugin in rviz2 to plan and execute tra
 robot as explained `here <https://moveit.picknik.ai/main/doc/tutorials/quickstart_in_rviz/quickstart_in_rviz_tutorial.html>`_.
 
 .. note::
-   Usually, MoveIt uses trajectory execution monitoring (TEM). If executing a trajectory takes too long,
-   e.g. because the action server died or the robot is blocked, this will lead to an error in
-   trajectory execution. However, this isn't necessarily compatible with the scaled joint
-   trajectory controller (as long as TEM is not aware of the scaling), as execution time can
-   be unbounded on purpose. Hence, TEM is disabled in the example MoveIt configuration:
+   The MoveIt configuration provided here has Trajectory Execution Monitoring (TEM) *disabled*, as the
+   Scaled Joint Trajectory Controller may cause trajectories to be executed at a lower velocity
+   than they were originally planned by MoveIt. MoveIt's TEM however is not aware of this
+   deliberate slow-down due to scaling and will in most cases unnecessarily (and unexpectedly)
+   abort goals.
+
+   Until this incompatibility is resolved, the default value for ``execution_duration_monitoring``
+   is set to ``false``. Users who wish to temporarily (re)enable TEM at runtime (for use with
+   other, non-scaling controllers) can do so using the ROS 2 parameter services supported by
+   MoveIt.
 
    .. literalinclude:: ../../ur_moveit_config/launch/ur_moveit.launch.py
       :language: python

--- a/ur_robot_driver/doc/usage.rst
+++ b/ur_robot_driver/doc/usage.rst
@@ -191,6 +191,21 @@ To test the driver with the example MoveIt-setup, first start the driver as desc
 Now you should be able to use the MoveIt Plugin in rviz2 to plan and execute trajectories with the
 robot as explained `here <https://moveit.picknik.ai/main/doc/tutorials/quickstart_in_rviz/quickstart_in_rviz_tutorial.html>`_.
 
+.. note::
+   Usually, MoveIt uses trajectory execution monitoring (TEM). If executing a trajectory takes too long,
+   e.g. because the action server died or the robot is blocked, this will lead to an error in
+   trajectory execution. However, this isn't necessarily compatible with the scaled joint
+   trajectory controller (as long as TEM is not aware of the scaling), as execution time can
+   be unbounded on purpose. Hence, TEM is disabled in the example MoveIt configuration:
+
+   .. literalinclude:: ../../ur_moveit_config/launch/ur_moveit.launch.py
+      :language: python
+      :start-at: trajectory_execution =
+      :end-at: execution_duration_monitoring": False
+      :append: }
+      :dedent: 4
+      :caption: ur_moveit_config/launch/ur_moveit.launch.py
+
 Fake hardware on ROS2 Galactic
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 

